torch/linalg/__init__.py

```diff
@@ -214,6 +214,86 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+cond = _add_docstr(_linalg.linalg_cond, r"""
+linalg.norm(input, p=None, *, out=None) -> Tensor
+
+Computes the condition number of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`, using the matrix norm defined by :attr:`p`.
+The condition number is defined as the matrix norm of
+:attr:`input` times the matrix norm of the inverse of :attr:`input`.
+
+This function supports ``float``, ``double``, and only on CPU, ``cfloat`` and ``cdouble`` dtype for :attr:`input`.
+
+.. note:: For `p = {None, 2, -2}` :attr:`input` can be non-square. For other norm types :attr:`input` must be
+          a square matrix or a batch of square matrices. If :attr:`input` does not satisfy this requirement
+          then a RuntimeError will be thrown.
+
+.. note:: If :attr:`input` is a non-invertible matrix then a tensor containing infinity will be returned.
+          If :attr:`input` is a batch of matrices and one or more of them is not invertible
```

Comment:
This may change. Throwing a runtime error is also fine, I'm curious to hear your thoughts.